lib/test/t_TensorProductExperiment_std.cxx

```diff
@@ -0,0 +1,234 @@
+//                         -*- C++ -*-
+/**
+ *  @brief The test file of class TensorProductExperiment for standard methods
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/OT.hxx"
+#include "openturns/OTtestcode.hxx"
+
+using namespace OT;
+using namespace OT::Test;
+
+// Check if two int are equal
+void assertEqual(int value1, int value2)
```

Comment:
I took it from `t_MarshallOlkinCopula_std.cxx`, that I developed during 2019's training sessions. I think it should, but this requires an update of the shared OTtestcode lib. Would that fit within coding rules @jschueller ?

Reply:
Looking at t_MarshallOlkinCopula_std.cxx, you need assertEqual, but with template arguments so as to avoid specialization for all types. Also it must be ignored from SWIG because in python we already have "assert x==y".
I'm not sur assertFalse & assertTrue are really needed.

Reply:
This is far away the present development, which is unrelated to the management of unit tests. I suggest to discuss that topic at the next TC.

Reply:
its silly to duplicate code, maybe cherry-pick my latest commit from misc branch ?
https://github.com/openturns/openturns/pull/1941/commits/cd39108fc0bafb10742f0cfb1be97dfc9664064c

Reply:
Ok. Thank you for the templated code. I will do that.

Reply:
Got it with:
```
$ git checkout master
$ git fetch upstream pull/1941/head:misc
$ git checkout tensorproduct
$ git cherry-pick cd39108fc0bafb10742f0cfb1be97dfc9664064c
```